src/pages/blog/starlingx-single-core-platform.md

```diff
@@ -0,0 +1,62 @@
+
+With the introduction of StarlingX 8.0, the default number of cores assigned to the platform for AIO deployments has been set to 1. This adjustment requires a careful balance to ensure that the performance remains unaffected despite having only one platform core. In the following sections, we will dive into some specific changes made to each service. By examining these modifications, we will gain a comprehensive understanding of how the system can operate in a single-core environment while preserving performance and stability.
+
+# Platform Adjustments
+
+## System Inventory
+
+System Inventory (sysinv) is a key component of StarlingX, responsible for maintaining a centralized and comprehensive inventory of hardware devices and system configuration data. The sysinv service plays a vital role in the overall orchestration and operation of the StarlingX infrastructure. The sysinv component demands significant computational resources, creating opportunities for optimization.
+
+### Periodic and Runtime Tasks
+
+Sysinv consists of three main services: sysinv-conductor, sysinv-agent and sysinv-api. These services are designed to monitor system changes at periodic time intervals and take appropriate action as they occur. Some of the sysinv-conductor and sysinv-agent periodic tasks have been redesigned. The primary changes involved refactoring legacy code, extending periodic time intervals, and breaking down large tasks into smaller ones. This allowed for the reassessment of individual time intervals and distribution of their execution over time, aiming to minimize CPU usage spikes.
+
+### Remote Procedure Calls (RPC)
+
+In previous versions, RPC communication between sysinv-api, sysinv-conductor and sysinv-agent processes relied on RabbitMQ as the communication transport layer. RabbitMQ uses a broker architecture responsible for delivering messages with a dedicated process, resulting in increased overhead. To avoid that, a new brokerless backend was developed using a library built on top of ZeroMQ for message queuing and MessagePack for data serialization.
+
+## Services Scale-Down
+
+Platform services are typically designed with threads or worker processes that are proportional to the number of available cores on the platform. However, simply reducing the number of cores to just one may not be sufficient, as each service also has a minimum scale requirement. The footprint of certain services was reduced seeking to find a balance between minimum scale and overall performance.
+
+This process of scaling down resource usage involves reducing the number of threads and workers assigned to services such as postgres, etcd, containerd, memcached, armada, keystone, barbican, docker-registry, docker-token-server, kube-apiserver, and kubelet. It is important to note that worker allocation rules remain the same, with the changes affecting only applications that run on single core.
```

Comment:
Is 'armada' the service from Airship or does it refer to a different tool?

Reply:
Since FluxCD is replacing Armada, I think we should remove that from the list here.